zerver/tests/test_auth_backends.py

```diff
@@ -544,6 +544,7 @@ def _mock_key(self: RateLimitedAuthenticationByUsername) -> str:
 
         def attempt_authentication(username: str, password: str) -> Optional[UserProfile]:
             request = HttpRequest()
+            request.session = self.client.session
```

Comment:
Since the `client` has nothing to do with what we're doing in this function, I think this should be set to a newly made instance of session rather than `client.session`.

Reply:
Used `Client().session` as that felt like the only better way. We cannot make `request=mock.MagicMock` as some rate limit code requires it to be a proper request. So, having a proper session felt better rather than using mock just for the session.

Reply:
We can use  a `MagicMock` for `request.session`, I think that'll be much less confusing than assigning `client.session` to it

Reply:
Thanks for demonstrating this @mateuszmandera 
I was looking [this file](https://github.com/django/django/blob/master/django/test/testcases.py#L155) and ended up concluding that's the better way. I should've had a good look into `django/tests/client.py`.

(changed to use `MagicMock`)